Assets/Scripts/LookingMobile.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LookingMobile : MonoBehaviour
+{
+    public Transform cameraTransform;
+    public float cameraSensitivity;
+
+    private Vector2 lookInput;
+    private float cameraPitch;
+
+    // Update is called once per frame
+    void Update()
+    {
+        Touch t = Input.GetTouch(0);
+        if (t.phase == TouchPhase.Moved)
+        {
+            lookInput = t.deltaPosition * cameraSensitivity * Time.deltaTime;
+            cameraPitch = Mathf.Clamp(cameraPitch - lookInput.y, -90f, 90f);
+            cameraTransform.localRotation = Quaternion.Euler(cameraPitch, 0, 0);
+            transform.Rotate(transform.up, lookInput.x);
+        }
+        else if (t.phase == TouchPhase.Stationary)
+        {
+            lookInput = Vector2.zero;
+        }
+        else if (Input.touchCount > 0 && Input.touches[0].phase == TouchPhase.Began)
+        {
+            Ray raycast = Camera.main.ScreenPointToRay(Input.GetTouch(0).position);
+            RaycastHit raycastHit;
+            if (Physics.Raycast(raycast, out raycastHit))
+            {
+                if (raycastHit.collider.CompareTag("Arrow"))
+                {
+                    ArrowMovement objectHit = raycastHit.collider.gameObject.transform.GetComponent<ArrowMovement>();
```

Comment:
This Stuff Should Be Moved Over The Arrow Script So Its OO

Reply:
It shouldnt for a few reasons. 

One the arrow script is attached to 20-30 game objects across the map. That would mean each arrow is checking in their update for a raycast hit. and that is alot of unnecessary duplication of code. The arrow script only needs to know the information. it doesnt need to actually teleport or check.

2nd the looking movement already uses the same touch system. and so if you were to put it in another script you would have multiple raycasts out. This script is not managing the arrows. It is simply checking if you have interacted with an arrow. Then teleports you to the location that arrow has in its script.

(Also I forgot to push a few last changes (like the comments you saw above. so am gonna push that now)